src/Symfony/Component/Validator/Constraints/NotBlank.php

```diff
@@ -28,4 +29,5 @@ class NotBlank extends Constraint
     );
 
     public $message = 'This value should not be blank.';
+    public $allowNull = false;
```

Comment:
should we consider deprecating it when not set to true?

Reply:
Why? It would annoy a lot of people, and you may expect that `null` is considered as blank. I would keep the default as is.

Reply:
IMHO we should follow https://docs.jboss.org/hibernate/stable/beanvalidation/api/javax/validation/constraints/NotBlank.html eventually for "NotBlank"

Reply:
"The annotated element must not be null and must contain at least one non-whitespace character"

Isn't it exactly the current behavior?

Reply:
hm you're right. I read it as "null elements are considered valid.", which is explicitly mentioned for other constraints, but not for NotBlank. It's confusing for having NotNull also.. :sweat_smile: 

moreover, currently it uses empty() whereas we should check for "one non-whitespace character". But that's a new topic :)

Until we start checking for whitespaced strings in NotBlank, NotBlank(allowsNull=true) will be the same as Length(min=1)

Reply:
> Until we start checking for whitespaced strings in NotBlank, NotBlank(allowsNull=true) will be the same as Length(min=1)

 Unfortunately no, `Length` always allows empty strings: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Constraints/LengthValidator.php#L33-L35

Don't ask me why 😅

Reply:
It's legit and logical. IT's the same for Email.
How would you say: It should be an empty string. But if it's not an empty string it should be of length(min = 10, max = 100) ?
Well with the current implementation it's easy: Just use Length.

Reply:
Legit, indeed. Logical I’m not so sure: a string of length 0 doesn’t have at least N chars...

Anyway it’s not related to this PR  and will definitely not change this behavior, it would be a huge BC break.